lib/enum_weasel/generators/tables/deletion.rb

```diff
@@ -0,0 +1,46 @@
+module EnumWeasel
+  class Generators
+    module Tables
+      class Deletion
+        attr_reader :model_enums, 
+          :database_enum_tables, 
+          :connection
+
+        def initialize(model_enums, database_enum_tables)
+          @model_enums = model_enums
+          @database_enum_tables = database_enum_tables
+          @connection = ActiveRecord::Base.connection
+        end
+
+        def call
+          old_enum_tables.each do |table|
+            delete_table table
+          end
+        end
+        
+        def old_enum_tables
+          database_enum_tables - formatted_model_enum_names
+        end
+
+        def delete_table enum_table
+          if connection.table_exists?(enum_table)
+            connection.drop_table enum_table 
+          end
+        end
+
+        def formatted_model_enum_names
```

Comment:
this is repeated in both classes may be worthwhile moving this into a parent class